kotlin/android/app/src/main/java/dev/firezone/android/core/data/Repository.kt

```diff
@@ -98,9 +111,25 @@ internal class Repository
 
         fun getDeviceIdSync(): String? = sharedPreferences.getString(DEVICE_ID_KEY, null)
 
-        fun getFavoritesSync(): HashSet<String> = HashSet(sharedPreferences.getStringSet(FAVORITE_RESOURCES_KEY, null).orEmpty())
+        private fun saveFavoritesSync() {
+            sharedPreferences.edit().putStringSet(FAVORITE_RESOURCES_KEY, favorites.value.inner).apply()
+            _favorites.value = Favorites(favorites.value.inner)
+        }
+
+        fun addFavoriteResource(id: String) {
+            favorites.value.inner.add(id)
+            saveFavoritesSync()
```

Comment:
Instead of repeating this, can we now subscribe to changes to the `MutableStateFlow` and save them on every change?

Reply:
Alternatively, we could make it directly a [`DataStore`](https://developer.android.com/topic/libraries/architecture/datastore) perhaps? Not sure how difficult that would be.

Reply:
It might be hard to get rid of because we're also manually triggering the flow to emit in `saveFavoritesSync`. Maybe in a future PR